securedrop/tests/functional/test_source_warnings.py

```diff
@@ -14,3 +14,11 @@ def test_warning_appears_if_tor_browser_not_in_use(self):
 
         assert ("We recommend using Tor Browser to access SecureDrop" in
                 warning_banner.text)
+
+        # User should be able to dismiss the warning
+        warning_dismiss_button = self.driver.find_element_by_id(
+            'use-tor-browser-close')
+        warning_dismiss_button.click()
+
+        # The banner should now be hidden
+        self.wait_for(lambda: warning_banner.is_displayed() is False)
```

Comment:
Note that:

        self.wait_for(lambda: True is False)

will pass the test. The **wait_for** function expects the function to raise an exception if it should wait some more. If it does not raise an exception, it will not wait even if the function returns False.

Reply:
doh, added an assert